src/server/session.ts

```diff
@@ -577,6 +578,35 @@ module ts.server {
             }, []);
         }
 
+        getSignatureHelpItems(line: number, offset: number, fileName: string): protocol.SignatureHelpItems {
+            var file = ts.normalizePath(fileName);
+            var project = this.projectService.getProjectForFile(file);
+            if (!project) {
+                throw Errors.NoProject;
+            }
+            
+            var compilerService = project.compilerService;
+            var position = compilerService.host.lineOffsetToPosition(file, line, offset);
+            var helpItems = compilerService.languageService.getSignatureHelpItems(file, position);
+            if (!helpItems) {
+                return undefined;
+            }
+            
+            var span = helpItems.applicableSpan;
+            var result:protocol.SignatureHelpItems = {
+                items: helpItems.items,
+                applicableSpan: {
+                    start: compilerService.host.positionToLineOffset(file, span.start),
+                    end: compilerService.host.positionToLineOffset(file, span.start + span.length)

```

Comment:
You guys seem to do something similar throughout the function, you might want to consider just making a helper that returns a `{start: number, end: number}`.